### PR TITLE
Tractogram scripts part7: segmentation scripts

### PR DESCRIPTION
**Pull request by EmmaRenauld**
# Quick description

The four segmentation scripts. (Renaming = made changes in many files). Improved doc everywhere to reference each other.

1) `scil_tractogram_segment_bundles_with_bundleseg.py`:
   - Was ok! Clarified verifications.
   
2) `scil_tractogram_segment_connections_from_labels.py`:
    - A lot of the lines have changed because I moved most of the processing to `scilpy.tractanalysis.connectivity_segmentation`! *Will be difficult to review, sorry*.
    - Added an option to save the final streamlines in out_dir. Previously, they were saved as soon as you added option `--out_dir`.
    - Added a lot of prints at debug level.
    - Only saving intermediate files if that step was processed.
    - Reusing the same variable (current_sft) rather than a separate sft name at each step to avoid multiplying data in memory. Using current_sft.streamlines rather than a separate streamlines = current_sft.streamlines, for the same reason.
    - Separating the method `remove_loops_and_sharp_turns` into two: remove_loops, remove_sharp turns for more clarification; the "remove_loops" part was used twice (option args.loop_max_angle), once at the remove loops step, once at the remove_sharp_turn steps. Unnecessary processing removed.
    - Encapsulating the "remove overlapping points" section in `scilpy.tractograms.streamline_operations`
    - Using the existing function "filter_streamlines_by_length" rather than the local _prune_length method.
  
3) `scil_tractogram_tractogram_segment_with_recobundles.py`:
    - Fixed the verbose. Still had a `if args.verbose`.
    - Explained more the space and origin management. Added a `to_rasmm`, `to_center` to clarify, but it's the default when loading.
    - Used new scilpy's save_tractogram, that checks if empty first.

4) `scil_tractogram_tractogram_segment_with_ROI_and_score.py`:
     - Considering that I already have a 30-script PR, let's do it in another PR!!

## Type of change

Check the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [x] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scilpy/tractanalysis/bundle_operations.py [hunk 1]
```diff
@@ -314,8 +314,8 @@ def outliers_removal_using_hierarchical_quickbundles(streamlines,
     return summary
 
 
-def remove_outliers(streamlines, threshold, nb_points=12, nb_samplings=30,
-                    fast_approx=False):
+def remove_outliers_qb(streamlines, threshold, nb_points=12, nb_samplings=30,
+                       fast_approx=False):
     """
     Wrapper to classify inliers and outliers from a list of streamlines.
```
Comment (karanphil) [inline scilpy/tractanalysis/bundle_operations.py:320]: Maybe add a reference to Quickbundles to justify the renaming "qb"?

**Review by karanphil — APPROVED** [review]
Looks good from my limited knowledge of these scripts! Only a small comment. Also, do you know why Github doesn't understand some of the scripts that were simply renamed, instead of thinking you deleted/added a file...

**Review by arnaudbore — COMMENTED** [review]
A couple of questions for this review.

**Review by CHrlS98 — APPROVED** [review]
I don't know much about segmentation script but I scrolled through the code and, as far as I'm concerned, it seems alright.

File: scripts/scil_tractogram_segment_connections_from_labels.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+    logging.info('*** Starting connection post-processing and saving. ***')
+    logging.info('    This can be long, be patient.')
+    time1 = time.time()
+    with h5py.File(args.out_hdf5, 'w') as hdf5_file:
+        construct_hdf5_header(hdf5_file, sft)
+        prune_length = not args.no_pruning
+        remove_loops = not args.no_remove_loops
+        remove_outliers = not args.no_remove_outliers
+        remove_curv_dev = not args.no_remove_curv_dev
+        construct_hdf5_from_connectivity(
```
[the remark this answers is not in the document]
Comment (EmmaRenauld) [reply scripts/scil_tractogram_segment_connections_from_labels.py:301]: I wondered the same. We are writing in our draft scilpy paper that we DON'T do that, and people can combine scripts themselves.

[the remark this answers is not in the document]
Comment (arnaudbore) [reply scripts/scil_tractogram_segment_connections_from_labels.py:301]: Would it make sense to have enough information to be able to replicate the preproc of this script and to do it outside of the script if the user wants and then use it without any preproc ?

[the remark this answers is not in the document]
Comment (frheault) [reply scripts/scil_tractogram_segment_connections_from_labels.py:301]: That would require 3-4 scripts to either support .h5 as input and output or to duplicate scripts (one version for trk/tck and one for h5).

That's a complexity that is not so worth it I think.

**Review by arnaudbore — APPROVED** [review]

**Pull request merged**